File: Latihan_B/php/Pc.php
<?php
include "Processor.php";
include "Disk.php";
include "Ram.php";

class Pc {
    private $procie;
    private $storage;
    private $memory;
    private $totalPrice;

    // constructor
    function __construct($procie, $storage, $memory, $totalPrice) {
        $this->procie = $procie;
        $this->storage = $storage;
        $this->memory = $memory;
        $this->totalPrice = $totalPrice;
    }

    // method set untuk procie, storage, memory, dan totalPrice
    function setProcie($procie) {
        $this->procie = $procie;
    }
    function setStorage($storage) {
        $this->storage = $storage;
    }
    function setMemory($memory) {
        $this->memory = $memory;
    }
    function setTotalPrice($totalPrice) {
        $this->totalPrice = $totalPrice;
    }

    // method get untuk procie, storage, memory, dan totalPrice
    function getProcie() {
        return $this->procie;
    }
    function getStorage() {
        return $this->storage;
    }
    function getMemory() {
        return $this->memory;
    }
    function getTotalPrice() {
        return $this->totalPrice;
    }

    // method untuk output Pc
    function printComputer() {
        $this->procie->printProcie();
        echo "============================". "<br/>";
        $this->storage->printDisk();
        echo "============================". "<br/>";
        $this->memory->printRam();
        echo "============================". "<br/>";
        echo "Total Price : ". "Rp. ". $this->getTotalPrice(). "<br/>";
    }

    // destructor
    function __destruct() {}
}
?>

File: Latihan_B/java/Pc.java
public class Pc
{
    private Processor procie;
    private Disk storage;
    private Ram memory;
    private int totalPrice;

    // constructor
    public Pc(){}
    public Pc(Processor procie, Disk storage, Ram memory, int totalPrice) {
        this.procie = procie;
        this.storage = storage;
        this.memory = memory;
        this.totalPrice = totalPrice;
    }

    // method set untuk procie, storage, memory, dan totalPrice
    public void setProcie(Processor procie) {
        this.procie = procie;
    }
    public void setStorage(Disk storage) {
        this.storage = storage;
    }
    public void setMemory(Ram memory) {
        this.memory = memory;
    }
    public void setTotalPrice(int totalPrice) {
        this.totalPrice = totalPrice;
    }

    // method get untuk procie, storage, memory, dan totalPrice
    public Processor getProcie() {
        return procie;
    }
    public Disk getStorage() {
        return storage;
    }
    public Ram getMemory() {
        return memory;
    }
    public int getTotalPrice() {
        return totalPrice;
    }

    // method output untuk Pc
    public void printComputer() {
        procie.printProcie();
        System.out.println("============================");
        storage.printDisk();
        System.out.println("============================");
        memory.printRam();
        System.out.println("============================");
        System.out.println("Total Price : " + "Rp. " + totalPrice);
    }
}
